workshop/content/20-typescript/30-hello-cdk/300-cdk-watch.md

```diff
@@ -0,0 +1,179 @@
++++
+title = "CDK Watch"
+weight = 300
++++
+
+## Faster Personal Deployments
+
+{{% notice info %}} This section is not necessary to complete the workshop, but we
+recommend that you take the time to see how `cdk watch` can speed up your personal
+deployments.
+{{% /notice %}}
+
+It's great that we have a working lambda! But what if you want to tweak the lambda
+code to get it just right? Let's say that we have now decided that we want our
+lambda function to respond with `"Good Morning, CDK!"` instead of `"Hello, CDK"`.
+
+So far, it seems like the only tool we have at our disposal to update our stack is
+`cdk deploy`. But `cdk deploy` takes time; it has to deploy your CloudFormation
+stack and upload the `lambda` directory from your disk to the boostrap bucket. If
+we're just changing our lambda code, we don't actually need to update the
+CloudFormation stack, so that part of `cdk deploy` is wasted effort.
+
+We really only need to update our lambda code. It would be great if we had
+some other mechanism for doing only that...
+
+## See how long 'CDK Deploy' takes
+
+First, as an exercise, change your lambda code in `lambda/hello.js` slightly and 
+time how long it takes to run `cdk deploy`. It will help us get a baseline for how 
+long `cdk deploy` takes:
+
+```bash
+time cdk deploy
+```
+
+The output will look something like this:
+
+```
+INSERT OUTPUT HERE
+```
+
+## CDK Watch
+
+{{% notice info %}} This command deliberately introduces drift in CloudFormation 
+stacks in order to speed up deployments. For this reason, only use it for 
+development purposes. Never use CDK Watch for your production deployments!
+{{% /notice %}}
+
+We can speed up that deployment time with `cdk watch`, which watches your
+CDK application and looks for hotswappable changes. Which `cdk watch` finds a
+change that can be hotswapped, it bypasses the CloudFormation deployment entirely
+and instead makes the necessary SDK call to update your code for you. Changes to 
+your AWS Lambda asset code, AWS Step Function State Machine definition, and other
+non-resource changes have the potential to be hotswapped. Here, we will try
+`cdk watch` with your AWS Lambda asset code.
```

Comment:
I would briefly mention the `--hotswap` flag here, and then go into more detail about how it works later on. I would change this section to something like:

"We can speed up that deploy time with `cdk watch`, which monitors your code and assets for changes and attempts to perform a deployment automatically when a change is detected. By default, `cdk watch` will use the `--hotswap` flag, which inspects the changes and determines if those changes can be hotswapped. If they can, the CDK CLI will use AWS service APIs to directly make the changes; otherwise it will fall back to performing a full CloudFormation deployment. To disable this behavior, pass `--no-hotswap` to `cdk watch`."

Reply:
I like Calvin's suggestion!